<a href="https://colab.research.google.com/github/GaabrielCoosta/SoulCodeAcademy/blob/main/C%C3%B3pia_de_Projeto_01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Dictionary of the company's HR base XPTO

In [ ]:
import pandas as pd

# Carregando o dicionário da base de dados da base de dados do RH
df = pd.read_csv('/content/drive/MyDrive/Projeto01-Colab_GitHub/Dicio.csv', sep=',', engine='python')

# tratando dos dados nulos
df = df.fillna('-')
print('Dicionário Base RH:')
df.head()



Dicionário Base RH:


Nome do Atributo                                      Significado  \
0               Idade                                            Idade   
1  Viagem de negócios  Frequência que o funcionário viaja pela empresa   
2   Diárias recebidas        Diárias recebidas pelos funcionários (R$)   
3        Departamento                  Departamento que está vinculado   
4   Distância de Casa          Distância de casa para o trabalho em km   

  Mais Informação.1 Mais Informação.2 Mais informação.3 Mais Informação.4  \
0                 -                 -                 -                 -   
1                 -                 -                 -                 -   
2                 -                 -                 -                 -   
3                 -                 -                 -                 -   
4                 -                 -                 -                 -   

  Mais Informação.5  
0                 -  
1                 -  
2                 -  
3                 -  
4                 -

# Database of employees who resigned due to attrition




In [ ]:
import pandas as pd

# Importando base de dados dos funcionários que pediram demissão
print('Base de dados dos funcionários que pediram demissão:')
basedesgaste = pd.read_csv('/content/drive/MyDrive/Projeto01-Colab_GitHub/basedesgaste.csv', sep=',', engine='python')

basedesgaste.head()

Idade  Estado Civil  Gênero  qtde de empresas que já trabalhou  \
0     41             0       0                                  8   
1     37             0       1                                  6   
2     28             0       1                                  5   
3     36             0       1                                  7   
4     34             0       1                                  2   

   Total de anos trabalhados  Total de anos trabalhados na empresa  \
0                          8                                     6   
1                          7                                     0   
2                          6                                     4   
3                         10                                     5   
4                          8                                     4   

   Satisfação com o trabalho  Nível de Satisfação com o ambiente de trabalho  \
0                          4                                               2   
1                          3                                               4   
2                          3                                               3   
3                          1                                               3   
4                          1                                               2   

   Satisfação nas relações no trabalho  Nível de envolvimento com o trabalho  \
0                                    1                                     3   
1                                    2                                     2   
2                                    2                                     2   
3                                    2                                     2   
4                                    3                                     3   

   ...  Anos trabalhados na função atual  Anos trabalhando com o atual gestor  \
0  ...                                 4                                    5   
1  ...                                 0                                    0   
2  ...                                 2                                    3   
3  ...                                 3                                    3   
4  ...                                 2                                    3   

   Anos desde a última promoção  \
0                             0   
1                             0   
2                             0   
3                             0   
4                             1   

   Qtde de treinamentos realizados no último ano  score de performance  \
0                                              0                     3   
1                                              3                     3   
2                                              4                     3   
3                                              4                     4   
4                                              2                     3   

   percentual de aumento de salário  Salário por hora  Diárias recebidas  \
0                                11          R$ 94,00        R$ 1.102,00   
1                                15          R$ 92,00        R$ 1.373,00   
2                                14          R$ 93,00          R$ 103,00   
3                                23          R$ 82,00        R$ 1.218,00   
4                                11          R$ 83,00          R$ 699,00   

  Salário Mensal BRUTO Renda mensal líquida  
0         R$ 19.479,00          R$ 5.993,00  
1          R$ 2.396,00          R$ 2.090,00  
2         R$ 12.947,00          R$ 2.028,00  
3          R$ 6.986,00          R$ 3.407,00  
4         R$ 17.102,00          R$ 2.960,00  

[5 rows x 22 columns]

# Descriptive Statistics and Dispersion
### Descriptive measures
- *Descriptive statistics organizes and describes data in three ways: through tables, graphs, and descriptive measures*

### Dispersion measures
- *Scattering measures help us understand how the data varies around the mean, i.e. they also help describe the trend in the data*







#1. Measures of Central Tendency

**Variable Descriptions:**

  *   *md = Média*
  *   *mediana = Mediana*
  *   moda = Moda

In [ ]:
import pandas as pd
# carregando os dados da tabela filtrada dos funciónarios que pediram demissão

basedesgaste = pd.read_csv('/content/drive/MyDrive/Projeto01-Colab_GitHub/desgastebase.csv', sep=',', engine='python')
#display(desgastados)

# Média
#print("-------- Média ---------")
md = basedesgaste.mean()
pd.set_option('display.float_format','{:.2f}'.format)
#display(md)


# Mediana
#print("-------- Mediana ---------")
mediana = basedesgaste.median()
pd.set_option('display.float_format','{:.2f}'.format)
#display(mediana)

# Moda
print("---------- Moda -----------")
moda = basedesgaste.mode()
moda = moda.dropna(how='all')
moda = moda.fillna('Unimodal')
#display(moda[0:2])



---------- Moda -----------


Idade Estado Civil    Gênero qtde de empresas que já trabalhou  \
0 29.00         0.00      1.00                              1.00   
1 31.00     Unimodal  Unimodal                          Unimodal   

  Total de anos trabalhados Total de anos trabalhados na empresa  \
0                      1.00                                 1.00   
1                  Unimodal                             Unimodal   

  Satisfação com o trabalho Nível de Satisfação com o ambiente de trabalho  \
0                      3.00                                           1.00   
1                  Unimodal                                       Unimodal   

  Satisfação nas relações no trabalho Nível de envolvimento com o trabalho  \
0                                3.00                                 3.00   
1                            Unimodal                             Unimodal   

   ... Anos trabalhados na função atual Anos trabalhando com o atual gestor  \
0  ...                             0.00                                0.00   
1  ...                         Unimodal                            Unimodal   

  Anos desde a última promoção Qtde de treinamentos realizados no último ano  \
0                         0.00                                          2.00   
1                     Unimodal                                      Unimodal   

  score de performance percentual de aumento de salário Salário por hora  \
0                 3.00                            11.00            66.00   
1             Unimodal                         Unimodal         Unimodal   

  Diárias recebidas Salário Mensal BRUTO  Renda mensal líquida  
0               240              9150.00               2293.00  
1               289             Unimodal               2362.00  

[2 rows x 22 columns]

#2. Dispersion Measures

**Variable descriptions:**

  *   *dp = Desvio Padrão*
  *   *va = Variância*
  *   *q1 = Quartil 1*
  *   *q3 = Quartil 3*
  *   *min = Mínimo valor*
  *   *max = Máximo valor*
  *   *amp = Amplitude Total*
  *   aiq = Amplitude Interquatilica


In [ ]:
import pandas as pd

# carregando os dados da tabela filtrada dos funciónarios que pediram demissão
basedesgaste = pd.read_csv('/content/drive/MyDrive/Projeto01-Colab_GitHub/desgastebase.csv', sep=',', engine='python')
#display(desgastados)

# Máximo
#print("O máximo dos atributos são:")
max = basedesgaste.max()
#display(max)

# Mínimo
min = basedesgaste.min()
#print("O mínimo dos atributos são:")
#display(min)

# Desvio Padrão
dp = basedesgaste.std()
#print("O desvio padrão dos atributos são:")
#display(dp)

# Variância
va = basedesgaste.var()
#print("A variância dos atributos são:")
#display(va)

# Amplitude Total
amp = basedesgaste.max() - basedesgaste.min()
#print("A amplitude total dos atributos são:")
#display(amp)

# Quartil 1
q1 = basedesgaste.quantile(.025)
#print("O quartil 1 dos atributos são:")
#display(md)

# Quartil 3
q3 = basedesgaste.quantile(0.75)
#print("O quartil 3 dos atributos são:")
#display(md)

# Amplitude Interquartil
aiq = basedesgaste.quantile(.75) - basedesgaste.quantile(.25)
#print("A amplitude interqualil dos atributos:")
#display(aiq)



Idade                                                58
Estado Civil                                          2
Gênero                                                1
qtde de empresas que já trabalhou                     9
Total de anos trabalhados                            40
Total de anos trabalhados na empresa                 40
Satisfação com o trabalho                             4
Nível de Satisfação com o ambiente de trabalho        4
Satisfação nas relações no trabalho                   4
Nível de envolvimento com o trabalho                  4
Equilíbrio vida trabalho                              4
Nível hierárquico                                     5
Anos trabalhados na função atual                     15
Anos trabalhando com o atual gestor                  14
Anos desde a última promoção                         15
Qtde de treinamentos realizados no último ano         6
score de performance                                  4
percentual de aumento de salário                

# Frequency Distributions


#Frequency of layoffs by hierarchical level


In [ ]:
import pandas as pd
n_frequencia = pd.read_csv('/content/drive/MyDrive/Projeto01-Colab_GitHub/frequenciah.csv', sep=',', engine='python')
n_frequencia = n_frequencia.fillna('-')
display(n_frequencia)

Nivel  Quantidade Quantidade acum % Quantidade % Quantidade acum
0      1         143          143.00       60,34%            60,34%
1      2          52          195.00       21,94%            82,28%
2      3          32          227.00       13,50%            95,78%
3      4           5          232.00        2,11%            97,89%
4      5           5          237.00        2,11%           100,00%
5  Total         237               -      100,00%                 -

- - In total, **237** employees resigned. **60.34%** had **level 1** and **21.94% level 2**, being the lowest levels of a scale from **1 to 5**. The sum of the two levels represent **82.28%** of worn out employees

# Frequency of layoffs by age

In [ ]:
import pandas as pd
idade = pd.read_csv('/content/drive/MyDrive/Projeto01-Colab_GitHub/idadecsv - Página1.csv', sep=',', engine='python')
idade = idade.fillna('-')
display(idade)

Frequência de demissão por idade  Unnamed: 1       Unnamed: 2    Unnamed: 3  \
0                            Idade  Quantidade  Quantidade acum  % Quantidade   
1                          18 - 25          44               44        18,57%   
2                          26 - 36         122              166        51,48%   
3                          37 - 45          37              203        15,61%   
4                          46 - 58          34              237        14,35%   
5                            Total         237                -       100,00%   

          Unnamed: 4  
0  % Quantidade acum  
1             18,57%  
2             70,04%  
3             85,65%  
4            100,00%  
5                  -

- - Through this analysis we obtained the data that **70.04%** of the worn employees were between **18 and 36** years old, and that **51.48%** were between **26** and **36** years old.

# Frequency of layoffs per time worked

In [ ]:
import pandas as pd
t_trabalho = pd.read_csv('/content/drive/MyDrive/Projeto01-Colab_GitHub/demissãoportempodetrabalho.csv', sep=',', engine='python')
t_trabalho = t_trabalho.fillna('-')
display(t_trabalho)

Frequência de demissão total de anos trabalhados na empresa  Unnamed: 1  \
0                                               Anos           Quantidade   
1                                              0 - 5                  162   
2                                               5-10                   55   
3                                            11 - 16                    8   
4                                             17- 22                    6   
5                                            23 - 28                    2   
6                                            29 - 34                    3   
7                                            35 - 40                    1   
8                                              Total                  237   

        Unnamed: 2    Unnamed: 3         Unnamed: 4  
0  Quantidade acum  % Quantidade  % Quantidade acum  
1              162        68,35%             68,35%  
2              217        23,21%             91,56%  
3              225         3,38%             94,94%  
4              231         2,53%             97,47%  
5              233         0,84%             98,31%  
6              236         1,27%             99,58%  
7              237         0,42%            100,00%  
8                -       100,00%                  -

- The majority of employees who resigned have less than **10 years** in the company, which includes **90.92%** of employees who resigned